### PR TITLE
Image scaling and Listbox Improvements

### PR DESCRIPTION
**Pull request by mjsmagalhaes**
- Improve image scaling to use the most restrictive scale.
  - Fix issues when the smaller dimention is the most restrictive one.
  - Clean up that code a bit.
- Increase the max number of colums in a listbox.

### CODE REVIEW

File:  crazys-wm-mod/CSurface.cpp [hunk 1]
```diff
@@ -264,53 +264,39 @@ bool CSurface::DrawSurface(int x, int y, SDL_Surface* destination, SDL_Rect* cli
...
-					else	// assume the height is larger
-						scaleX = scaleY = ((double)clip->h / (double)m_Surface->h);
-					m_Temp = zoomSurface(m_Surface, scaleX, scaleY, 1);
-				}
-			}
-			else
-			{
-				if (clip->w != m_Surface->w || clip->h != m_Surface->h)
+				scaleX = ((double)clip->w / (double)m_Surface->w);
+				scaleY = ((double)clip->h / (double)m_Surface->h);
```
Comment (aevojoey) [inline  crazys-wm-mod/CSurface.cpp:270]: Need to add a checks to prevent divide by 0 errors

File:  crazys-wm-mod/CSurface.cpp [hunk 2]
```diff
@@ -264,53 +264,39 @@ bool CSurface::DrawSurface(int x, int y, SDL_Surface* destination, SDL_Rect* cli
...
 			{
-				if (clip->w != m_Surface->w && clip->h != m_Surface->h)
-				{
-					if (m_Surface->w > m_Surface->h)	// if the width is larger so scale down based on the width but keep aspect ratio
-						scaleX = scaleY = ((double)clip->w / (double)m_Surface->w);
-					else	// assume the height is larger
-						scaleX = scaleY = ((double)clip->h / (double)m_Surface->h);
-					m_Temp = zoomSurface(m_Surface, scaleX, scaleY, 1);
-				}
-			}
```
Comment (aevojoey) [inline  crazys-wm-mod/CSurface.cpp:277]: You have two possible return failures, the ifs need else statements or they may not do anything.

Comment (mjsmagalhaes) [reply  crazys-wm-mod/CSurface.cpp:277]: That is the idea ... If they don't do anything the following code will use the existing m_Temp or the m_Surface.

File:  crazys-wm-mod/CSurface.cpp (hunk 1 above)
Comment (mjsmagalhaes) [reply  crazys-wm-mod/CSurface.cpp:270]: I've just updated the scaling algorithm. Everything else is the same with just some polish.

Perhaps it is easier if you add the check after commiting those changes.

**Pull request merged**